test/unit/org/apache/cassandra/db/guardrails/GuardrailAlterTableTest.java

```diff
@@ -56,29 +56,29 @@ public void testAlterTableGuardrail() throws Throwable
         setGuardrail(true);
         assertValid("ALTER TABLE %s WITH compression = { 'class' : 'SnappyCompressor', 'chunk_length_in_kb' : 32 };");
 
-        // Confirm works when disabled
+        // Confirm guardrail works when disabled
         setGuardrail(false);
-        assertValid("ALTER TABLE %s ADD test_one text;");
+        assertFails("ALTER TABLE %s ADD test_one text;", "changing columns");
 
-        // Confirm fails when enabled
+        // Confirm works when enabled
         setGuardrail(true);
-        assertFails("ALTER TABLE %s ADD test_two text;", "changing columns");
+        assertValid("ALTER TABLE %s ADD test_two text;");
 
-        // Confirm works again when disabled
+        // Confirm fails again when disabled
         setGuardrail(false);
-        assertValid("ALTER TABLE %s ADD test_three text;");
+        assertFails("ALTER TABLE %s ADD test_three text;", "changing columns");
 
-        // Fail to ALTER-DROP
-        setGuardrail(true);
+        // Confirm stops ALTER + DROP columns
+        setGuardrail(false);
         assertFails("ALTER TABLE %s DROP test_one", "changing columns");
 
-        // Fail to ALTER-RENAME
+        // Confirm stops ALTER + RENAME columns
         assertFails("ALTER TABLE %s RENAME test_one TO test_ren_one", "changing columns");
 
-        // Allow ALTER table options even when alter table is disabled
+        // Allow ALTER table *options* works even when alter table is disabled
         assertValid("ALTER TABLE %s WITH GC_GRACE_SECONDS = 123; ");
 
-        // Restore things so they work for other tests
+        // Restore things so they work for other tests that may be added to this class in the future
         setGuardrail(true);
```

Comment:
Maybe better in an @After or in a `try/finally` block?

Reply:
try/finally, yep. Will change.